core/cap-0059.md

```diff
@@ -0,0 +1,343 @@
+## Preamble
+
+```
+CAP: 0059
+Title: Host functions for BLS12-381
+Working Group:
+    Owner: Jay Geng <@jayz22>
+    Authors: Jay Geng <@jayz22>
+    Consulted: TBD
+Status: Draft
+Created: 2024-08-15
+Discussion: https://github.com/stellar/stellar-protocol/discussions/1500
+Protocol version: 22
+```
+
+## Simple Summary
+BLS12-381 is a pairing-friendly elliptic curve which enables a new suite of applications. This CAP proposes a set of new host functions providing access to BLS12-381 curve operations.
+
+## Working Group
+
+As described in the preamble section.
+
+## Motivation
+Pairing friendly elliptic curve operations are the backbone of many privacy-preserving applications, such as BLS signature, zkSNARKs, and anonymous login. BLS12-381 is one of the most well-known and widely adopted pairing friendly curves, due to its efficiency and 128-bit security. However, the curve operations are inherently computationally intensive, making them prohibitive to be implemented in a smart contract. Providing an efficient implementation for a well-chosen set of curve and field operations natively inside the Soroban host is crucial for unlocking privacy-preserving applications in smart contracts.
+
+### Goals Alignment
+This CAP is aligned with the following Stellar Network Goals:
+
+  - The Stellar Network should make it easy for developers of Stellar projects
+    to create highly usable products
+
+## Abstract
+11 new host functions are proposed for performing field and curve operations on BLS12-381. Definitions and the semantics of each new host function, as well as the associated new metering parameters, will be introduced and explained. 
+
+## Specification
+
```

Comment:
Should we specify whether points on the curves used in the host functions are in uncompressed(x and y coordinates) or compressed (x coordinate) form? 
From XDR changes it looks like we are using uncompressed form.

Reply:
done